### ISSUE TITLE
dagit - when we exhaust user watches constraint on Linux we end up in a bad state

### ISSUE THREAD

**Issue by natekupp** [dagster-io/dagster#1682]
Attempting to run dagit w/ the default # of user watches set:

```
(venv) ubuntu@ip-172-31-7-247:/opt/dagster$ /opt/dagster/venv/bin/python /opt/dagster/venv/bin/dagit -h 0.0.0.0 -p 3000 -y /opt/dagster/app/repository.yaml
Will be watching for file changes...
Traceback (most recent call last):
  File "/opt/dagster/venv/bin/dagit", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/opt/dagster/dagster/python_modules/dagit/bin/dagit", line 72, in <module>
    observer.start()
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/api.py", line 255, in start
    emitter.start()
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/utils/__init__.py", line 110, in start
    self.on_thread_start()
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify.py", line 121, in on_thread_start
    self._inotify = InotifyBuffer(path, self.watch.is_recursive)
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify_buffer.py", line 35, in __init__
    self._inotify = Inotify(path, recursive)
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify_c.py", line 199, in __init__
    self._add_dir_watch(path, recursive, event_mask)
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify_c.py", line 387, in _add_dir_watch
    self._add_watch(full_path, mask)
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify_c.py", line 401, in _add_watch
    Inotify._raise_error()
  File "/opt/dagster/venv/lib/python3.5/site-packages/watchdog/observers/inotify_c.py", line 413, in _raise_error
    raise OSError("inotify watch limit reached")
OSError: inotify watch limit reached
(venv) ubuntu@ip-172-31-7-247:/opt/dagster$ Loading repository...
Serving on http://0.0.0.0:3000
```

Running
```
echo "fs.inotify.max_user_watches=524288" | sudo tee -a /etc/sysctl.conf
sudo sysctl -p
```

Fixes this issue

**Comment by natekupp** [comment]
Ah, @mgasner pointed out this might be because of `observer.schedule(handler, '.', True)` in `dagit` 

if a repository yaml is specified, we should watch just that directory, instead of `'.'`

**Comment by natekupp** [comment]
e.g. the invocation above is `/opt/dagster/venv/bin/python /opt/dagster/venv/bin/dagit -h 0.0.0.0 -p 3000 -y /opt/dagster/app/repository.yaml`

**Comment by natekupp** [comment]
I think this is resolved by the fixes to #1684 - will reopen if I see it again
